server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/model/Condition.java

```diff
@@ -75,6 +75,27 @@ public final int hashCode() {
         }
     }
 
+    class MatchNothingCondition implements Condition {
+        public Operator.OperatorName getOperator() {
+            return Operator.OperatorName.MATCH_ALL;
+        }
+
+        @Override
+        public boolean matches(String line) {
+            return false;
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            return o instanceof MatchNothingCondition;
```

Comment:
use the standard way of doing it

Reply:
Sorry, the standard equals hashCode is only applicable with a classing having fields... I don't have any field in that class and I expect every instances are equal

Reply:
anyway, I removed it :)